include/deal.II/matrix_free/hanging_nodes_internal.h

```diff
@@ -113,6 +113,8 @@ namespace internal
      */
     enum ConstraintTypes : unsigned int
     {
+      default_value = 0,
+
```

Comment:
@kronbichler I have introduced here a default value. Else I got some warnings in the tests in structures like this:
```
(direction == 0) ? dealii::internal::MatrixFreeFunctions::ConstraintTypes::face_y : 0;
```
Is the name fine or do you prefer another name?

Reply:
I guess this does not necessarily mean that we have no constraints at all (in which case I would have chosen `unconstrained`), but only that we do not add new content through a binary 'or' operation, right? In that case, I am fine with the name.

Reply:
I must admit I like `unconstrained`, since this is the way we would use it in most cases (and we could replace all the comparisons with 0). The concrete instance that caused trouble looks like this:

```
            const bool constrained_face =
              (constraint_mask[v] &
               (((direction == 0) ? dealii::internal::MatrixFreeFunctions::
                                      ConstraintTypes::face_y :
                                    dealii::internal::MatrixFreeFunctions::
                                      ConstraintTypes::default_value) |
                ((direction == 1) ? dealii::internal::MatrixFreeFunctions::
                                      ConstraintTypes::face_x :
                                    dealii::internal::MatrixFreeFunctions::
                                      ConstraintTypes::default_value)));
```

Reply:
But then we go with `unconstrained`?

Reply:
Yup. I have renamed it!